govcd/metadata_test.go

```diff
@@ -181,3 +182,165 @@ func (vcd *TestVCD) Test_DeleteMetadataOnVm(check *C) {
 		}
 	}
 }
+
+func (vcd *TestVCD) Test_DeleteMetadataOnVAppTemplate(check *C) {
+
+	fmt.Printf("Running: %s\n", check.TestName())
+	cat, err := vcd.org.FindCatalog(vcd.config.VCD.Catalog.Name)
+	if err != nil {
+		check.Skip("Test_DeleteMetadataOnCatalogItem: Catalog not found. Test can't proceed")
+	}
+
+	if vcd.config.VCD.Catalog.CatalogItem == "" {
+		check.Skip("Test_DeleteMetadataOnCatalogItem: Catalog Item not given. Test can't proceed")
+	}
+
+	catItem, err := cat.FindCatalogItem(vcd.config.VCD.Catalog.CatalogItem)
+	check.Assert(err, IsNil)
+
+	vAppTemplate, err := catItem.GetVAppTemplate()
+	check.Assert(err, IsNil)
+
+	// Add metadata
+	_, err = vAppTemplate.AddMetadata("key2", "value2")
+	check.Assert(err, IsNil)
+
+	// Remove metadata
+	err = vAppTemplate.DeleteMetadata("key2")
+	check.Assert(err, IsNil)
+
+	metadata, err := vAppTemplate.GetMetadata()
+	check.Assert(err, IsNil)
```

Comment:
Should check whether metadata is valid

Reply:
nothing to check, later lines check if there is metadata still deleted